python/ray/data/_internal/stats.py

```diff
@@ -310,94 +328,216 @@ def _summarize_iter(self) -> str:
-            memory_stats = [
-                round(e.max_rss_bytes / (1024 * 1024), 2) for e in exec_stats
-            ]
             out += "* Peak heap memory usage (MiB): {} min, {} max, {} mean\n".format(
-                min(memory_stats),
-                max(memory_stats),
-                int(np.mean(memory_stats)),
+                memory_stats["min"],
+                memory_stats["max"],
+                memory_stats["mean"],
             )
 
-        output_num_rows = [m.num_rows for m in blocks if m.num_rows is not None]
-        if output_num_rows:
+        output_num_rows_stats = stage_stats.get(DatasetStats.OUTPUT_NUM_ROWS, {})
+        if output_num_rows_stats:
             out += indent
             out += "* Output num rows: {} min, {} max, {} mean, {} total\n".format(
-                min(output_num_rows),
-                max(output_num_rows),
-                int(np.mean(output_num_rows)),
-                sum(output_num_rows),
+                output_num_rows_stats["min"],
+                output_num_rows_stats["max"],
+                output_num_rows_stats["mean"],
+                output_num_rows_stats["sum"],
             )
 
-        output_size_bytes = [m.size_bytes for m in blocks if m.size_bytes is not None]
-        if output_size_bytes:
+        output_size_bytes_stats = stage_stats.get(DatasetStats.OUTPUT_SIZE_BYTES, {})
+        if output_size_bytes_stats:
             out += indent
             out += "* Output size bytes: {} min, {} max, {} mean, {} total\n".format(
-                min(output_size_bytes),
-                max(output_size_bytes),
-                int(np.mean(output_size_bytes)),
-                sum(output_size_bytes),
+                output_size_bytes_stats["min"],
+                output_size_bytes_stats["max"],
+                output_size_bytes_stats["mean"],
+                output_size_bytes_stats["sum"],
             )
 
-        if exec_stats:
-            node_counts = collections.defaultdict(int)
-            for s in exec_stats:
-                node_counts[s.node_id] += 1
+        node_count_stats = stage_stats.get(DatasetStats.NODE_COUNT, {})
+        if node_count_stats:
             out += indent
             out += "* Tasks per node: {} min, {} max, {} mean; {} nodes used\n".format(
-                min(node_counts.values()),
-                max(node_counts.values()),
-                int(np.mean(list(node_counts.values()))),
-                len(node_counts),
+                node_count_stats["min"],
+                node_count_stats["max"],
+                node_count_stats["mean"],
+                node_count_stats["count"],
             )
-
         return out
 
+    def get_total_wall_time(self) -> float:
+        total_wall_time = 0
+        for stage_name, ss in self.stage_stats.items():
+            stage_wall_time_stats = ss.get(DatasetStats.WALL_TIME)
+            if stage_wall_time_stats:
+                total_wall_time += stage_wall_time_stats.get("sum", 0)
```

Comment:
Walltime should be "max"? It may not be useful for "sum" of parallel tasks' wall times.

Reply:
I am thinking we should keep both. Sum of all tasks wall times is useful to get throughput (sum_total_time / total_input_data_size), and also the total cluster resource being used.

Max could help figure out the straggler.

Reply:
The throughput is the (end_walltime - start_walltime) / total_input_size, and end_walltime - start_walltime is calculated with the sum of max (not sum of sum), right?

Reply:
Sorry right, throughput is total_input_size / (end_walltime - start_walltime). But resource usage should also include IO besides CPU time. Cool, I think we can change to max_wall_time here.

Reply:
Got it, based on the discussion above, I will replace `get_total_wall_time` with `get_max_wall_time`.

Reply:
Hmm, for naming, could we keep using "total"? The "total" is referring to the sum of walltimes across all stages. We only need to make sure the walltime of each stage is the max walltime across all blocks. (i.e. from SUM(SUM(walltime across blocks) to SUM(MAX(walltime across blocks)), the outer SUM is what "total" referring to).

Reply:
Right, that makes sense thanks. I have implemented it using the sum of max wall times per stage.